### DISCUSSION TITLE
Arranging child node structure?

### DISCUSSION THREAD

**Discussion by manumoi** [freeplane/freeplane#501] [Help]
Hello I m new to freeplane , and thanks a lot for the software. Very cool so far.

I have used extensively mindmanager in the past, and one feature i found really helpful for readability was the possibility to rearrange the way childs of a specific node would be displayed (eg, horizontally, vertically with parent in the middle, vertically with parent on top, etc). There are multiple arrangement styles/options in mindmanager. You can mix such styles in the same map, and it helps the readability of the map a lot, according to its context of use.
I ve been looking for a similar feature in freeplane but could not find it. Does it exist? If yes, how to do it? If not, could this feature be considered in future releases?

**Comment by manumoi** [comment]
Sorry but nobody has an answer?

**Comment by dpolivaev** [comment]
The feature does not exist in Freeplane. I am not sure if it can be easily implemented.

**Reply by manumoi** [reply]
Thanks. I actually just noticed a discussion (https://github.com/freeplane/freeplane/discussions/503) referring to one of the node layout arrangements that is actually also available on MindManager. Perhaps you could have a look at what their team did, for inspiration (e.g. with google image). Having the ability to arrange trees of child nodes in different ways is very helpful for structuring ideas and it definitely improves readability. But i understand implementing such an improved UI could be challenging if it has not be thought about from start. Thanks for the software :)

**Reply by euu2021** [reply]
I'm curious to see how mindmanager shows the node (that is not the root) with branches on both sides. I mean, like the orange one in this drawing:
![IMG_20220805_170739](https://user-images.githubusercontent.com/77707706/183154909-8d75131d-dd64-4c56-8a80-13d414693dc2.jpg)

Is it possible to do that with layouts in mindmanager?

**Reply by manumoi** [reply]
wasnt sure but was able to access a version of mind manager 8. So your map is not possible (which makes sense because it would be a nightmare to further expand the left nodes of your red submap). 
So with this old version i had, i made a quick graph showing most of the node layout possibilities of mind manager 8
![possible layouts in mind manager](https://user-images.githubusercontent.com/80925887/183173535-848e6021-e399-496b-9f98-f8fe35b1d50c.jpg)
Here, the layout option of the root node for its children, is the map layout. For non root nodes, as i wrote, you can only choose between 3 layouts for their children (map, tree, org-chart). But you can switch again at any time in the hierarchy as i showed in the map.

**Reply by manumoi** [reply]
actually little adjustments to my comments. I was wondering if layout options for children of non-root nodes would be different if the layout of the root node was not map. 
So layout options do not change, but compared to details i wrote:
- tree will always place child nodes below the parent node
- org chart seem to place child nodes in the same direction of the main map (i.e. in a map where the root node has a up org chart for its children, child nodes  of a non-root node arranged in a org chart layout will be above the parent node)
- map  will always place childe nodes on the right of the parent node
This test (see below) is made with the root node having the up org chart layout for its children. It s possible behaviours are also different for other layouts of the root nod
![possible layouts in mind manager2](https://user-images.githubusercontent.com/80925887/183207597-54feaeca-df41-4a6f-b0ea-30588b77ef64.jpg)
e

**Reply by Illegit** [reply]
I see the possibility to apply a script to a node children ( I do not have enough knowledge maybe,  but for sure no time to try it...) and to create a script for each kind of solution with some parameters to size it a little bit... 
Maybe on the basis of this one..
[createMatrixLink.zip](https://github.com/freeplane/freeplane/files/9273830/createMatrixLink.zip)
Just in case of...
